stl/inc/xspan

```diff
@@ -0,0 +1,238 @@
+
+namespace details {
+    template <size_t _Extent>
+    class _Extent_type {
+    public:
+        using index_type = size_t;
+
+        constexpr _Extent_type() noexcept = default;
+
+        template <index_type _OtherExtent>
+        constexpr _Extent_type(_Extent_type<_OtherExtent> _Ext) noexcept {
+            static_assert(_OtherExtent == _Extent || _OtherExtent == dynamic_extent,
+                "Mismatch between static extent and size of initializing data.");
+#if _CONTAINER_DEBUG_LEVEL != 0
+            _STL_VERIFY(_Ext.size() == _Extent, "Mismatch between static extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr _Extent_type([[maybe_unused]] index_type _Size) noexcept {
+#if _CONTAINER_DEBUG_LEVEL != 0
+            _STL_VERIFY(_Size == _Extent, "Mismatch between dynamic extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr index_type size() const noexcept {
+            return _Extent;
+        }
+    };
+
+    template <>
+    class _Extent_type<dynamic_extent> {
+    public:
+        using index_type = size_t;
+
+        constexpr _Extent_type() noexcept : _Mysize(index_type(0)) {}
+
+        template <index_type _OtherExtent>
+        explicit constexpr _Extent_type(_Extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext.size()) {}
+
+        explicit constexpr _Extent_type(index_type _Size) noexcept : _Mysize(_Size) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Mysize >= 0, "Mismatch between dynamic extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr index_type size() const noexcept {
+            return _Mysize;
+        }
+
+    private:
+        index_type _Mysize;
+    };
+} // namespace details
+
+// [views.span], class template span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span {
+public:
+    using element_type    = _Ty;
+    using value_type      = remove_cv_t<_Ty>;
+    using index_type      = size_t;
+    using difference_type = ptrdiff_t;
+    using pointer         = _Ty*;
+    using const_pointer   = const _Ty*;
+    using reference       = _Ty&;
+    using const_reference = const _Ty&;
+    // using iterator               = _Span_iterator<pointer>;
+    // using const_iterator         = _Span_const_iterator<const_pointer>;
+    // using reverse_iterator       = _STD reverse_iterator<iterator>;
+    // using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    // [span.cons], span constructors, copy, assignment, and destructor
+    constexpr span() noexcept : _Mypair(_Zero_then_variadic_args_t(), nullptr) {}
+
+    constexpr span(pointer _Ptr, index_type _Count) noexcept : _Mypair(_One_then_variadic_args_t(), _Count, _Ptr) {}
+
+    constexpr span(pointer _First, pointer _Last) noexcept
+        : _Mypair(_One_then_variadic_args_t(), static_cast<index_type>(distance(_First, _Last)), _First) {}

```

Comment:
All calls to non-`_Ugly`, non-member functions need to be `_STD` qualified. `distance(_First, _Last)` can be "hijacked" by ADL, while `_STD distance(_First, _Last)` always calls the Standard function. Occurs below (e.g. `addressof`), needs an exhaustive audit.

Reply:
Thanks, uglyfication is still more or less a mystery